Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIScriptingCocoa.mm

```diff
@@ -0,0 +1,188 @@
+                    [webView _evaluateJavaScript:script.value().first withSourceURL:script.value().second.value() inFrame:info inContentWorld:executionWorld.wrapper() completionHandler:makeBlockPtr([info = RetainPtr { info }, numberOfExecutions, results = WTFMove(results), callback = WTFMove(callback)](id resultOfJSExecution, NSError *error) mutable {
+                        addResultAndCompleteIfDone(resultOfJSExecution, info.get(), error.localizedDescription, numberOfExecutions, WTFMove(results), WTFMove(callback));
+                    }).get()];
+                }
+                
+                return;
+            }
+
+            NSArray *arguments = parameters.args ? createNSArray(parameters.args.value()).get() : @[ ];
+            NSString *function = [NSString stringWithFormat:@"return (%@)(...args)", (NSString *)(parameters.func.value())];
+
+            [webView _callAsyncJavaScript:function arguments:@{ @"args": arguments } inFrame:info inContentWorld:executionWorld.wrapper() completionHandler:makeBlockPtr([info = RetainPtr { info }, numberOfExecutions, results = WTFMove(results), callback = WTFMove(callback)](id resultOfJSExecution, NSError *error) mutable {
+                addResultAndCompleteIfDone(resultOfJSExecution, info.get(), error.localizedDescription, numberOfExecutions, WTFMove(results), WTFMove(callback));
+            }).get()];
+        }
+    }).get()];
+}
+
+void WebExtensionContext::scriptingInsertCSS(WebPageProxyIdentifier identifier, const WebExtensionScriptInjectionParameters& parameters, CompletionHandler<void()>&& callback)
+{
+    WKWebView *webView = m_tabMap.get(parameters.identifier.value())->mainWebView();
+    if (!hasPermission(webView.URL)) {
+        callback();
+        return;
+    }
+
+    // FIXME: <https://webkit.org/b/262491> Add API to allow for css injections in specific frames.
+
+    auto page = WebProcessProxy::webPage(identifier);
+    WebUserContentControllerProxy& controller = page.get()->userContentController();
+
+    auto injectedFrames = parameters.allFrames ? WebCore::UserContentInjectedFrames::InjectInAllFrames : WebCore::UserContentInjectedFrames::InjectInTopFrameOnly;
+
+    Vector<std::optional<SourcePair>> styleSheetPairs;
+    populateStyleSheetPairs(WTFMove(styleSheetPairs), m_extension, parameters.files, parameters.css);
+
+    for (auto& styleSheet : styleSheetPairs) {
+        auto userStyleSheet = API::UserStyleSheet::create(WebCore::UserStyleSheet { styleSheet.value().first, styleSheet.value().second.value_or(URL(""_s)), Vector<String>(), Vector<String>(), injectedFrames, WebCore::UserStyleUserLevel, std::nullopt }, *m_contentScriptWorld);
+
+        controller.addUserStyleSheet(userStyleSheet);
+        m_dynamicallyInjectedUserStyleSheets.append(userStyleSheet);
+    }
+
+    callback();
+}
+
+void WebExtensionContext::scriptingRemoveCSS(WebPageProxyIdentifier identifier, const WebExtensionScriptInjectionParameters& parameters, CompletionHandler<void()>&& callback)
```

Comment:
`CompletionHandler<void()>&& completionHandler`

Also should these take a `std::options<String> error` for when the tab isn't found, etc?